reporter/reporter.go

```diff
@@ -84,23 +109,30 @@ func NewReporter(
 ) Reporter {
 	clockOpts := opts.ClockOptions()
 	instrumentOpts := opts.InstrumentOptions()
-	return &reporter{
+	r := &reporter{
 		matcher:         matcher,
 		server:          server,
 		nowFn:           clockOpts.NowFn(),
 		maxPositiveSkew: clockOpts.MaxPositiveSkew(),
 		maxNegativeSkew: clockOpts.MaxNegativeSkew(),
-		metrics:         newReporterMetrics(instrumentOpts.MetricsScope(), instrumentOpts.MetricsSamplingRate()),
+		reportInterval:  instrumentOpts.ReportInterval(),
+		doneCh:          make(chan struct{}),
+		metrics:         newReporterMetrics(instrumentOpts),
 	}
+
+	r.wg.Add(1)
```

Comment:
Perhaps worth moving this into a `Run` method to decouple constructing the reporter from reporting metrics?

Reply:
I thought about it, it would require adding the `Run` method to the interface, and my main worry is that callers would forget to call `Run` in which case the reporter would still work but "mysteriously" not reporting metrics.

I'd be more in favor of adding the `Run` method if what the `Run` method would be doing is something that's more complicated than starting up a goroutine for metrics reporting, but in this case it seems pretty innocuous and `Close` method will wait for the goroutine to exit so it won't leak either. Wdyt?

Reply:
Sounds good to me. My generally stance is to avoid hidden goroutines, but that's just because they are easy to leak. Given that we have a Close method and leak detector tests I don't see why we need to change the interface.